packages/mds-agency/request-handlers.ts

```diff
@@ -137,39 +152,46 @@ export const getVehiclesByProvider = async (req: AgencyApiRequest, res: AgencyAp
 
   try {
     const response = await getVehicles(skip, take, url, req.query, provider_id)
-    return res.status(200).send(response)
+    const { version } = res.locals
+    return res.status(200).send({ version, ...response })
   } catch (err) {
     logger.error('getVehicles fail', err)
-    return res.status(500).send(new ServerError())
+    return res.status(500).send({
+      error: 'server_error',
+      error_description: 'Unknown server error'
+    })
   }
 }
 
 export async function updateVehicleFail(
   req: AgencyApiRequest,
-  res: AgencyApiResponse,
+  res: AgencyUpdateVehicleResponse,
   provider_id: UUID,
   device_id: UUID,
   err: Error | string
 ) {
   if (String(err).includes('not found')) {
     res.status(404).send({
-      error: 'not_found'
+      error: 'not_found',
+      error_description: 'Vehicle not found'
     })
   } else if (String(err).includes('invalid')) {
     res.status(400).send({
-      error: 'invalid_data'
+      error: 'invalid_data',
+      error_description: 'Invalid parameters for vehicle were sent'
```

Comment:
Looks like we ought to be using `bad_param` here https://github.com/openmobilityfoundation/mobility-data-specification/tree/dev/agency#vehicle---update